### ISSUE TITLE
How to improve the strength of denoising by adjusting parameters?

### ISSUE THREAD

**Issue by LAccordeur** [HomeOfVapourSynthEvolution/VapourSynth-BM3D#22]
I had tried to change the sigma, the radius, the profile and other parameters, but these didn't work on some noise videos. Can you give me some advice?

**Comment by mysteryx93** [comment]
Personally I haven't found BM3D to be ajustable at all in terms of denoising strength. Sigma=9 is what always worked best in my tests.

To increase denoising strength, one trick is to use SMDegrain as a first pass and pass it as the ref parameter. [That's what xClean is doing.](https://forum.doom9.org/showthread.php?t=183245)

**Comment by mawen1250** [comment]
Basically "sigma" works as a threshold for frequency filtering.
It seems  the strength of the noise is way higher above the threshold, so you may need to try much higher "sigma".

For very strong noise with temporal fluctuation, pure BM3D may not work very well.
As @mysteryx93 mentioned, passing a pre-filtered clip as "ref" may help. (while personally I haven't found a good enough filter at the time I wrote BM3D)

**Comment by mawen1250** [comment]
> Personally I haven't found BM3D to be ajustable at all in terms of denoising strength. Sigma=9 is what always worked best in my tests.
> 
> To increase denoising strength, one trick is to use SMDegrain as a first pass and pass it as the ref parameter. [That's what xClean is doing.](https://forum.doom9.org/showthread.php?t=183245)

Indeed, I also found that I constantly set sigma=8 whenever I use BM3D for denoising.
This might be because it's a good trade-off between noise reduction and over-smoothness.

**Comment by mysteryx93** [comment]
What I found is that sigma changes the sharpness of the edges; with little or no impact on removed noise. Anything below 9 (or 8) is "not enough" and anything above 9 gives too sharp edges, from my tests.
